### PR TITLE
DSEGOG-251 DSEGOG-252 Improvements to the false colour

### PR DESCRIPTION
**Pull request by joshdimanteto**
## Description
Enter a description of the changes here

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
closes DSEGOG-251, closes DSEGOG-252

### CODE REVIEW

**Comment by joshdimanteto** [comment]
false colour

File: e2e/images.spec.ts-snapshots/user-can-change-the-false-colour-to-use-reverse-1-chromium-linux.png
Comment (louise-davies) [inline e2e/images.spec.ts-snapshots/user-can-change-the-false-colour-to-use-reverse-1-chromium-linux.png:1]: This screenshot doesn't match the others or the colourbar so it was probably taken too early I'd guess? Let's wait until Kenneth's playwright stuff is merged and fix it after that

File: e2e/traces.spec.ts-snapshots/user-can-change-trace-via-clicking-on-a-thumbnail-1-webkit-linux.png
Comment (louise-davies) [inline e2e/traces.spec.ts-snapshots/user-can-change-trace-via-clicking-on-a-thumbnail-1-webkit-linux.png:1]: This snapshot shouldn't have changes - it now no longer matches the other browsers. Again, let's wait for the playwright docker stuff

File: src/images/falseColourPanel.component.tsx
Comment (louise-davies) [inline src/images/falseColourPanel.component.tsx]: What is the point of this custom component? Might as well just use `ListSubheader`

File: src/images/falseColourPanel.component.tsx
Comment (louise-davies) [inline src/images/falseColourPanel.component.tsx]: These variables should probably be renamed to have the prefix `extended` rather than `main`

File: src/images/falseColourPanel.component.tsx
Comment (louise-davies) [inline src/images/falseColourPanel.component.tsx]: Could this code not be achieved with a loop over `mainColourMapTypeNames`/`mainColourMapNames` to reduce code repetition? Or just a simple for loop, and access both arrays by index.

Comment (louise-davies) [reply src/images/falseColourPanel.component.tsx]: Also, if this is then done by a loop, then there wouldn't need to be two separate components for `ColourMapMain` and `ColourMapExtended` - the difference between them would just be that the main one is only supplied with one item to "loop" over

File: src/images/falseColourPanel.component.tsx
Comment (louise-davies) [inline src/images/falseColourPanel.component.tsx]: This function can be extracted outside the component - a slight performance optimisation

File: src/images/falseColourPanel.component.tsx
Comment (joshdimanteto) [reply src/images/falseColourPanel.component.tsx]: The menuItems do not work correctly when rendered in a loop. When I initially did it a loop the menuItems were not clickable.

File: src/images/falseColourPanel.component.tsx
Comment (louise-davies) [inline src/images/falseColourPanel.component.tsx]: We can get rid of all this styling if we remove the `track={false}` bit - as MUI already has the indicators of the selected vs non-selected range. This greying out stuff was needed for the separate sliders to indicate that the min slider must be below the max slider and vice versa, but with them both on the same track this is no longer necessary and we can use the in-built MUI styling

**Review by louise-davies — CHANGES_REQUESTED** [review]
A few comments, and some stuff with the playwright tests that needs to wait for Kenneth's work. But it is working nicely :)

File: src/images/falseColourPanel.component.tsx
Comment (louise-davies) [reply src/images/falseColourPanel.component.tsx]: The issue is you need to ensure that the `MenuItem`s are direct children of the `Select`, which can be done like so:

```
{mainColourMapNames.map((mapNames, index) => {
        return [
          <MyListSubheader>{mainColourMapTypeNames[index]}</MyListSubheader>,
          mapNames.map((colourMap) => (
            <MenuItem key={colourMap} value={colourMap}>
              {colourMap}
            </MenuItem>
          )),
        ];
      })}
```

Here, we ensure we're returning arrays, which React parses to make every item a direct descendant.

Comment (joshdimanteto) [reply src/images/falseColourPanel.component.tsx]: Actually I've got it working using a loop. The Select component was complaining about me using react.Fragment before so I've just implemented a loop without using React.fragment

**Review by louise-davies — APPROVED** [review]
LGTM - review comments have been addressed and e2e tests passing now using snapshots generated by docker

**Pull request merged**
